ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -580,7 +620,18 @@ let dss_mem_vms doms =
       in
       let other_ds =
         if domid = 0 then
-          None
+          match mem_available () with
+          | Ok mem ->
+              Some
+                ( Rrd.VM uuid
+                , Ds.ds_make ~name:"memory_internal_free" ~units:"B"
+                    ~description:"Dom0 current free memory"
+                    ~value:(Rrd.VT_Int64 mem) ~ty:Rrd.Gauge ~min:0.0
+                    ~default:true ()
+                )
+          | Error exn ->
```

Comment:
I think `exn` is the abbreviation of `exception`.

Reply:
`exn` does not have to be an exception - it is just a string; you could name it `msg` (message) and it would be as valid. We simply want to log it.

Reply:
Avoid confusion, change `exn` to `msg `